requirements.txt

```diff
@@ -1,18 +1,28 @@
+# Core
```

Comment:
@barrywhart @barrywhart I went to use `requirements.txt` to take advantage of layer caching but found it was missing `typing_extensions`. I've copied the deps from `setup.py`.
Another example of the requirements getting out of sync so will pull together a PR to move `setup.py -> setup.cfg` and remove `requirements.txt`.